presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveDynamicPartitionPruning.java

```diff
@@ -0,0 +1,219 @@
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertEquals(dynamicFiltersStats.getDynamicFiltersCompleted(), 1L);
+
+        DynamicFilterDomainStats domainStats = getOnlyElement(dynamicFiltersStats.getDynamicFilterDomainStats());
+        assertEquals(domainStats.getSimplifiedDomain(), Domain.create(ValueSet.ofRanges(
+                range(INTEGER, 1L, true, 100L, true)), false)
+                .toString(getSession().toConnectorSession()));
+        assertEquals(domainStats.getDiscreteValuesCount(), 0);
+        assertEquals(domainStats.getRangeCount(), 1);
+    }
+
+    @Test(timeOut = 30_000)
+    public void testJoinWithMultipleDynamicFiltersOnProbe()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        // supplier names Supplier#000000001 and Supplier#000000002 match suppkey 1 and 2
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                getSession(),
+                "SELECT * FROM (" +
+                        "SELECT supplier.suppkey FROM " +
+                        "partitioned_lineitem JOIN tpch.tiny.supplier ON partitioned_lineitem.suppkey = supplier.suppkey AND supplier.name IN ('Supplier#000000001', 'Supplier#000000002')" +
+                        ") t JOIN tpch.tiny.partsupp ON t.suppkey = partsupp.suppkey AND partsupp.suppkey IN (2, 3)");
+        assertGreaterThan(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "tpch:" + PARTITIONED_LINEITEM);
+        // Probe-side is partially scanned
+        assertLessThanOrEqual(probeStats.getInputPositions(), 615L);
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), 0L);
+
+        DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(result.getQueryId());
+        assertEquals(dynamicFiltersStats.getTotalDynamicFilters(), 2L);
+        assertEquals(dynamicFiltersStats.getLazyDynamicFilters(), 2L);
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertEquals(dynamicFiltersStats.getDynamicFiltersCompleted(), 2L);
+
+        List<DynamicFilterDomainStats> domainStats = dynamicFiltersStats.getDynamicFilterDomainStats();
+        assertEquals(domainStats.size(), 2L);
+        domainStats.forEach(stats -> {
+            assertGreaterThanOrEqual(stats.getRangeCount(), 1);
+            assertEquals(stats.getDiscreteValuesCount(), 0);
+        });
+    }
+
+    private DynamicFiltersStats getDynamicFilteringStats(QueryId queryId)
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        return runner.getCoordinator()
+                .getQueryManager()
+                .getFullQueryInfo(queryId)
+                .getQueryStats()
+                .getDynamicFiltersStats();
+    }
+
+    private long countRows(String tableName)
```

Comment:
please add a test case where build side is too large, see: `io.prestosql.plugin.memory.TestMemorySmoke#testJoinLargeBuildSideNoDynamicFiltering`

Reply:
Added